### PR TITLE
Ascend NPU add topk 

### PR DESCRIPTION
**Pull request by OleNet**
### PR types

New features
### PR changes

OPs
### Describe

[NPU] ops for TopKD

image

### CODE REVIEW

File: paddle/fluid/operators/top_k_op_npu.cc [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  }
+};
+
+}  // namespace operators
+}  // namespace paddle
+
+namespace ops = paddle::operators;
+
+REGISTER_OP_NPU_KERNEL(
+    top_k,
```
Comment (zhiqiu) [inline paddle/fluid/operators/top_k_op_npu.cc:87]: only fp16?

File: python/paddle/fluid/tests/unittests/npu/test_top_k_op_npu.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+paddle.enable_static()
+SEED = 2021
+
+
+@unittest.skipIf(not paddle.is_compiled_with_npu(),
+                 "core is not compiled with NPU")
+class TestTopk(OpTest):
+    def setUp(self):
+        self.set_npu()
+        self.place = paddle.NPUPlace(4)
```
Comment (zhiqiu) [inline python/paddle/fluid/tests/unittests/npu/test_top_k_op_npu.py]: Better set to 0

File: python/paddle/fluid/tests/unittests/npu/test_top_k_op_npu.py [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+    def test_check_output(self):
+        self.check_output_with_place(self.place, check_dygraph=False)
+
+
+@unittest.skipIf(not paddle.is_compiled_with_npu(),
+                 "core is not compiled with NPU")
+class TestTopkV2(OpTest):
+    def setUp(self):
+        self.set_npu()
+        self.place = paddle.NPUPlace(4)
```
Comment (zhiqiu) [inline python/paddle/fluid/tests/unittests/npu/test_top_k_op_npu.py]: Same

File: paddle/fluid/operators/top_k_op_npu.cc [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+ public:
+  void Compute(const framework::ExecutionContext& ctx) const override {
+    // read input
+    auto* input = ctx.Input<framework::LoDTensor>("X");
+    auto* output = ctx.Output<framework::LoDTensor>("Out");
+    auto* indices = ctx.Output<framework::LoDTensor>("Indices");
+
+    size_t k = static_cast<int>(ctx.Attr<int>("k"));
+
+    output->mutable_data<paddle::platform::float16>(ctx.GetPlace());
```
Comment (liym27) [inline paddle/fluid/operators/top_k_op_npu.cc]: `mutable_data<paddle::platform::float16>`  -> `mutable_data<T>`

File: paddle/fluid/operators/top_k_op_npu.cc [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    indices->mutable_data<int>(ctx.GetPlace());
+
```
Comment (liym27) [inline paddle/fluid/operators/top_k_op_npu.cc:53]: The indices data type  must be `int64`.

**Review by liym27 — APPROVED** [review]
LGTM

**Pull request merged**